pytorch_lightning/trainer/trainer.py

```diff
@@ -581,6 +583,8 @@ def run_train(self) -> None:
 
         self.checkpoint_connector.has_trained = False
 
+        reset_seed()
+
```

Comment:
adding this here is the fix

Reply:
Can we move this into `run_sanity_check` instead ? It would be cleaner. Mind adding a comment too ?

Reply:
> Can we move this into `run_sanity_check` instead ? It would be cleaner. Mind adding a comment too ?

Reply:
good call. changed!